internal/datasystem/fdv1_datasystem.go

```diff
@@ -0,0 +1,146 @@
+package datasystem
+
+import (
+	"github.com/launchdarkly/go-server-sdk/v7/interfaces"
+	"github.com/launchdarkly/go-server-sdk/v7/internal"
+	"github.com/launchdarkly/go-server-sdk/v7/internal/datasource"
+	"github.com/launchdarkly/go-server-sdk/v7/internal/datastore"
+	"github.com/launchdarkly/go-server-sdk/v7/ldcomponents"
+	"github.com/launchdarkly/go-server-sdk/v7/subsystems"
+)
+
+type FDv1 struct {
+	dataSourceStatusBroadcaster *internal.Broadcaster[interfaces.DataSourceStatus]
+	dataSourceStatusProvider    interfaces.DataSourceStatusProvider
+	dataStoreStatusBroadcaster  *internal.Broadcaster[interfaces.DataStoreStatus]
+	dataStoreStatusProvider     interfaces.DataStoreStatusProvider
+	flagChangeEventBroadcaster  *internal.Broadcaster[interfaces.FlagChangeEvent]
+	dataStore                   subsystems.DataStore
+	dataSource                  subsystems.DataSource
+}
+
+func NewFDv1(dataStoreFactory subsystems.ComponentConfigurer[subsystems.DataStore], dataSourceFactory subsystems.ComponentConfigurer[subsystems.DataSource], clientContext *internal.ClientContextImpl) (*FDv1, error) {
+	system := &FDv1{
```

Comment:
This is basically a copy-paste of existing SDK configuration logic. If it looks like a bunch of circular dependencies, that's because it is.